lib/logger.js

```diff
@@ -222,22 +222,21 @@ Logger.prototype._bufferLog = function(message) {
 
     if (!this._flusher) {
         debug('No scheduled flush. Scheduling for %d ms from now.', configs.FLUSH_INTERVAL);
-        this._flusher = setTimeout(bind(this._flush, this, (err) => {
-            if (err) {
-                debug('Received an error while flushing...' + err);
+        this._flusher = setTimeout(() => {
+            if (this._buf.length > 0) {
+                this._flush((e) => {debug(e);});
```

Comment:
I feel like the logging here begins to lose context, because it seems like you will just end up with a bunch of text like `Unsuccessful request` or `An error occured while making this request` which I'm not sure it really helps with actual debugging.

Not to say what was there was better, but this could be made better.

Reply:
Added more context.